src/Microsoft.ML.ImageAnalytics/Imager.cs

```diff
@@ -0,0 +1,420 @@
+        {
+            if (imagePixelData.Length != width * height * 4)
+            {
+                throw new ArgumentException($"Invalid {nameof(imagePixelData)} buffer size.");
+            }
+
+            SKBitmap image = new SKBitmap(new SKImageInfo(width, height, SKColorType.Bgra8888));
+
+            Debug.Assert(image.Info.BitsPerPixel == 32);
+            Debug.Assert(image.RowBytes * image.Height == width * height * 4);
+
+            imagePixelData.CopyTo(new Span<byte>(image.GetPixels().ToPointer(), image.Width * image.Height * 4));
+
+            return new Imager(image);
+        }
+
+        /// <summary>
+        /// Gets the image pixel data and how the colors are ordered in the used pixel format.
+        /// </summary>
+        /// <param name="alphaIndex">The index of the alpha in the pixel format.</param>
+        /// <param name="redIndex">The index of the red color in the pixel format.</param>
+        /// <param name="greenIndex">The index of the green color in the pixel format.</param>
+        /// <param name="blueIndex">The index of the blue color in the pixel format.</param>
+        /// <returns>The byte span containing the image pixel data.</returns>
+        public ReadOnlySpan<byte> Get32bbpImageData(out int alphaIndex, out int redIndex, out int greenIndex, out int blueIndex)
+        {
+            ThrowInvalidOperationExceptionIfDisposed();
+
+            if (_image.Info.ColorType == SKColorType.Rgba8888)
+            {
+                redIndex = 0;
+                greenIndex = 1;
+                blueIndex = 2;
+                alphaIndex = 3;
+            }
+            else
+            {
+                Debug.Assert(_image.Info.ColorType == SKColorType.Bgra8888);
+                blueIndex = 0;
+                greenIndex = 1;
+                redIndex = 2;
+                alphaIndex = 3;
+            }
+
+            Debug.Assert(_image.Info.BytesPerPixel == 4);
+
+            return _image.GetPixelSpan();
+        }
+
+        internal Imager CloneWithResizing(int width, int height, ImageResizeMode mode)
+        {
+            ThrowInvalidOperationExceptionIfDisposed();
+
+            SKBitmap image = mode switch
+            {
+                ImageResizeMode.Pad => ResizeWithPadding(width, height),
+                ImageResizeMode.Fill => ResizeFull(width, height),
+                >= ImageResizeMode.CropAnchorTop and <= ImageResizeMode.CropAnchorCentral => ResizeWithCrop(width, height, mode),
+                _ => throw new ArgumentException($"Invalid resize mode value.", nameof(mode))
+            };
+
+            if (image is null)
+            {
+                throw new InvalidOperationException($"Couldn't resize the image");
+            }
+
+            return new Imager(image);
+        }
+
+        private SKBitmap ResizeFull(int width, int height) => _image.Resize(new SKSizeI(width, height), SKFilterQuality.None);
+
+        private SKBitmap ResizeWithPadding(int width, int height)
```

Comment:
Do you need to define the color value for the padding?

Reply:
No, ML never supported specifying the padding. It is black. Note, these are private methods which we can extend or do more if we find more scenarios need that.

Reply:
Ok 👍